crates/cheatcodes/src/lib.rs

```diff
@@ -109,18 +107,36 @@ impl<T: Cheatcode> DynCheatcode for T {
 }
 
 /// The cheatcode context, used in [`Cheatcode`].
-pub(crate) struct CheatsCtxt<'a, 'b, DB: DatabaseExt> {
+pub(crate) struct CheatsCtxt<'cheats, 'evm, DB: DatabaseExt> {
     /// The cheatcodes inspector state.
-    pub(crate) state: &'a mut Cheatcodes,
+    pub(crate) state: &'cheats mut Cheatcodes,
     /// The EVM data.
-    pub(crate) ecx: &'b mut EvmContext<DB>,
+    pub(crate) ecx: &'evm mut InnerEvmContext<DB>,
+    /// The precompiles context.
+    pub(crate) precompiles: &'evm mut ContextPrecompiles<DB>,
```

Comment:
I've inlined `EvmContext` because otherwise it invokes the deref on self.ecx which doesn't allow multiple mutable borrows (e.g. ecx.db and ecx.journaled_state would be in conflict)